edalize/vivado.py

```diff
@@ -97,6 +97,22 @@ def __init__(self, edam=None, work_root=None, eda_api=None, verbose=True):
     def configure_main(self):
         self.vivado.configure()
 
+    def run_main(self):
+        """
+        Program the FPGA.
+
+        For programming the FPGA a vivado tcl script is written that searches for the
+        correct FPGA board and then downloads the bitstream. The tcl script is then
+        executed in Vivado's batch mode.
+        """
+        if "pnr" in self.tool_options:
+            if self.tool_options["pnr"] == "vivado":
```

Comment:
I'm not sure I understand what this code does. Do we need it?

Reply:
It seems it was originally added as an option to only do synthesis: f9f1bb9df694541750a942ccaf45ac6c0031bec3

Reply:
Ah, of course. That prevents it from trying to program iif we didn't do P&R.